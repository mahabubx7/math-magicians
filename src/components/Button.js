import PropTypes from 'prop-types';

function Button({ content, className }) {
  return (
    <button type="button" className={className}>{content}</button>
  );
}

Button.defaultProps = {
  className: '',
};

Button.propTypes = {
  content: PropTypes.string.isRequired,
  className: PropTypes.string,
};

export default Button;
