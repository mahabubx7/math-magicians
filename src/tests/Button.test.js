import { fireEvent, render, screen } from '@testing-library/react';
import Button from '../components/Button';

describe('Tests for Button component', () => {
  test('should render the Button component', () => {
    const mockFn = jest.fn((x) => 7 + x);
    render(<Button content="7" handle={mockFn} />);
    expect(screen.getByRole('button')).toBeInTheDocument();
    expect(screen.getByRole('button')).toHaveTextContent('7');
  });

  test('should be able to handle events', () => {
    let x = 0;
    const mockFn = jest.fn(() => {
      x += 1;
      return x;
    });
    render(<Button content="7" handle={mockFn} />);
    fireEvent.click(screen.getByRole('button'));
    expect(x).toBe(1);
  });

  test('should match the snapshot', () => {
    const mockFn = jest.fn((x) => 7 + x);
    render(<Button content="7" handle={mockFn} />);
    expect(screen.getByRole('button')).toMatchSnapshot('<button type="button" class="">7</button>');
  });
});
