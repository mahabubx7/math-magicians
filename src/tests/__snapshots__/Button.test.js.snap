// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Button component should match the snapshot: <button type="button" class="">7</button> 1`] = `
<button
  class=""
  type="button"
>
  7
</button>
`;
